wgpu-hal/src/gles/egl.rs

```diff
@@ -1082,7 +1082,8 @@ impl crate::Instance for Instance {
             unsafe { gl.debug_message_callback(super::gl_debug_message_callback) };
         }
 
-        // Avoid accidental drop when the context is not current.
+        // Wrap in ManuallyDrop to make it easier to "current" the
+        // GL context before dropping this GLOW context.
         let gl = ManuallyDrop::new(gl);
         inner.egl.unmake_current();
```

Comment:
This comment was trying to point out why we wrap in `ManuallyDrop` before calling `unmake_current`. I feel like the new comment is more about the role `ManuallyDrop` plays in the drop implementation of `AdapterContext`. While these are somewhat linked, I think it would be valuable for the comment here to focus specifically on the relation to the `unmake_current` call.

Perhaps something like:
```rs
// The GL context must be current when dropping `glow::Context`. Before `unmake_current`, wrap in
// `ManuallyDrop` to prevent unexpected panics from dropping this while the context is not current.
```

Reply:
@Imberflur we already discussed in https://github.com/gfx-rs/wgpu/pull/6114#discussion_r1733498062 that this comment is going to change.

Reply:
Ah okay! I wasn't sure if that was still being done so I added this comment 😅